velox/vector/BaseVector.h

```diff
@@ -863,6 +863,18 @@ class BaseVector {
     storageByteCount_ = std::nullopt;
   }
 
+  // Slice a buffer with specific type.
+  //
+  // For boolean type and if the offset is not multiple of 8, return a shifted
+  // copy; otherwise return a BufferView into the original buffer (with shared
+  // ownership of original buffer).
+  static BufferPtr sliceBuffer(
+      const Type&,
```

Comment:
please, add argument names and clarify that 'pool' is used to allocate new buffer for bool case.